### PR TITLE
ref(types): Add request types to integrations/

### PR DESCRIPTION
**Pull request by mgaeta**
See https://github.com/getsentry/sentry/pull/30732.

### CODE REVIEW

File: src/sentry/api/endpoints/project_filter_details.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ def put(self, request, project, filter_id):
         new_state = inbound_filters.set_filter_state(filter_id, project, serializer.validated_data)
         audit_log_state = AuditLogEntryEvent.PROJECT_ENABLE
 
+        returned_state = None
```
Comment (NisanthanNanthakumar) [inline src/sentry/api/endpoints/project_filter_details.py:37]: @mgaeta is this change intentional?

File: src/sentry_plugins/asana/plugin.py [hunk 2]
```diff
@@ -146,31 +147,31 @@ def error_message_from_json(self, data):
...
+    def create_issue(self, request: Request, group, form_data, **kwargs):
         client = self.get_client(request.user)
 
         try:
             response = client.create_issue(
                 workspace=self.get_option("workspace", group.project), data=form_data
             )
         except Exception as e:
-            self.raise_error(e, identity=client.auth)
+            raise self.raise_error(e, identity=client.auth)
```
Comment (NisanthanNanthakumar) [inline src/sentry_plugins/asana/plugin.py:158]: @mgaeta self.raise_error raises the error itself. I don't think we need this `raise` here.

Comment (mgaeta) [reply src/sentry_plugins/asana/plugin.py:158]: pyright doesn't understand the `NoReturn` return type so its flow analysis needs this hint to avoid the `reportUnboundVariable` error.

Comment (NisanthanNanthakumar) [reply src/sentry_plugins/asana/plugin.py:158]: NIT: In that case, I think `return` makes more sense, since the error is raised by the function.

**Review by NisanthanNanthakumar — APPROVED** [review]
